### PR TITLE
Added Forms Community Toolkit animations code

### PR DESCRIPTION
**Pull request by jsuarezruiz**
Fixes # .
- None

Changes Proposed in this pull request:
- Animations library code (Sample will be added in other PR).

### CODE REVIEW

File: src/Animations/Animations/Animations.csproj [hunk 1]
```diff
@@ -37,32 +37,49 @@
...
+  <Import Project="..\..\packages\Xamarin.Forms.2.3.4.247\build\portable-win+net45+wp80+win81+wpa81+MonoAndroid10+Xamarin.iOS10+xamarinmac20\Xamarin.Forms.targets" Condition="Exists('..\..\packages\Xamarin.Forms.2.3.4.247\build\portable-win+net45+wp80+win81+wpa81+MonoAndroid10+Xamarin.iOS10+xamarinmac20\Xamarin.Forms.targets')" />
   <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
-      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
+      <ErrorText>Este proyecto hace referencia a los paquetes NuGet que faltan en este equipo. Use la restauración de paquetes NuGet para descargarlos. Para obtener más información, consulte http://go.microsoft.com/fwlink/?LinkID=322105. El archivo que falta es {0}.</ErrorText>
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations.csproj]: Best to keep this in English

File: src/Animations/Animations/Animations/Base/AnimationBase.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System.Threading.Tasks;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/Base/AnimationBase.cs]: usings outside namepace declaration

File: src/Animations/Animations/Animations/Base/AnimationBase.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    using Xamarin.Forms;
+
+    public abstract class AnimationBase : BindableObject
+    {
+        public static readonly BindableProperty TargetProperty =
+            BindableProperty.Create("Target", typeof(VisualElement), typeof(AnimationBase), null,
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/Base/AnimationBase.cs]: nameof(Target), idem for both properties below

File: src/Animations/Animations/Animations/BounceAnimation.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/BounceAnimation.cs]: usings outside namepace declaration

File: src/Animations/Animations/Animations/FadeAnimation.cs [hunk 5]
```diff
@@ -0,0 +1,128 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/FadeAnimation.cs]: usings outside namepace declaration

File: src/Animations/Animations/Animations/FadeAnimation.cs [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    using System.Threading.Tasks;
+    using Xamarin.Forms;
+
+    public class FadeToAnimation : AnimationBase
+    {
+        public static readonly BindableProperty OpacityProperty =
+            BindableProperty.Create("Opacity", typeof(double), typeof(FadeToAnimation), default(double),
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/FadeAnimation.cs]: nameof, idem below

File: src/Animations/Animations/Animations/FlipAnimation.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/FlipAnimation.cs]: usings outside namepace declaration

File: src/Animations/Animations/Animations/FlipAnimation.cs [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    public class FlipAnimation : AnimationBase
+    {
+        public enum FlipDirection
+        {
+            Left,
+            Right
+        }
+
+        public static readonly BindableProperty DirectionProperty = 
+            BindableProperty.Create("Direction", typeof(FlipDirection), typeof(FlipAnimation), FlipDirection.Right,
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/FlipAnimation.cs]: nameof

File: src/Animations/Animations/Animations/HeartAnimation.cs [hunk 9]
```diff
@@ -0,0 +1,57 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/HeartAnimation.cs]: usings outside namepace declaration

File: src/Animations/Animations/Animations/JumpAnimation.cs [hunk 10]
```diff
@@ -0,0 +1,49 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/JumpAnimation.cs]: usings outside namepace declaration

File: src/Animations/Animations/Animations/RotateAnimation.cs [hunk 11]
```diff
@@ -0,0 +1,98 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/RotateAnimation.cs]: usings outside namepace declaration

File: src/Animations/Animations/Animations/RotateAnimation.cs [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+    using System.Threading.Tasks;
+    using Xamarin.Forms;
+
+    public class RotateToAnimation : AnimationBase
+    {
+        public static readonly BindableProperty RotationProperty = 
+            BindableProperty.Create("Rotation", typeof(double), typeof(RotateToAnimation), 0,
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/RotateAnimation.cs]: nameof, idem below

File: src/Animations/Animations/Animations/ScaleAnimation.cs [hunk 13]
```diff
@@ -0,0 +1,52 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/ScaleAnimation.cs]: usings outside namepace declaration

File: src/Animations/Animations/Animations/ScaleAnimation.cs [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+    using System.Threading.Tasks;
+    using Xamarin.Forms;
+
+    public class ScaleToAnimation : AnimationBase
+    {
+        public static readonly BindableProperty ScaleProperty =
+            BindableProperty.Create("Scale", typeof(double), typeof(ScaleToAnimation), 0,
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/ScaleAnimation.cs]: nameof, idem below

File: src/Animations/Animations/Animations/ShakeAnimation.cs [hunk 15]
```diff
@@ -0,0 +1,74 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/ShakeAnimation.cs]: usings outside namepace declaration

File: src/Animations/Animations/Animations/StoryBoard.cs [hunk 16]
```diff
@@ -0,0 +1,36 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System.Collections.Generic;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/StoryBoard.cs]: usings outside namepace declaration

File: src/Animations/Animations/Animations/TranslateAnimation.cs [hunk 17]
```diff
@@ -0,0 +1,39 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/TranslateAnimation.cs]: usings outside namepace declaration

File: src/Animations/Animations/Animations/TranslateAnimation.cs [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+    using System.Threading.Tasks;
+    using Xamarin.Forms;
+
+    public class TranslateToAnimation : AnimationBase
+    {
+        public static readonly BindableProperty TranslateXProperty =
+            BindableProperty.Create("TranslateX", typeof(double), typeof(TranslateToAnimation), default(double),
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/TranslateAnimation.cs]: nameof

File: src/Animations/Animations/Animations/TurnstileAnimation.cs [hunk 19]
```diff
@@ -0,0 +1,64 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/TurnstileAnimation.cs]: usings outside namepace declaration

File: src/Animations/Animations/Behaviors/BeginAnimationBehavior.cs [hunk 20]
```diff
@@ -0,0 +1,44 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System.Threading.Tasks;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Behaviors/BeginAnimationBehavior.cs]: usings outside namepace declaration

File: src/Animations/Animations/Behaviors/BeginAnimationBehavior.cs [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+        protected override void OnDetachingFrom(VisualElement bindable)
+        {
+            associatedObject = null;
+            base.OnDetachingFrom(bindable);
+        }
+
+        public static readonly BindableProperty AnimationProperty =
+            BindableProperty.Create("Animation", typeof(AnimationBase), typeof(BeginAnimationBehavior), null,
```
Comment (bartlannoeye) [inline src/Animations/Animations/Behaviors/BeginAnimationBehavior.cs]: nameof

File: src/Animations/Animations/Extensions/AnimationExtension.cs [hunk 22]
```diff
@@ -0,0 +1,24 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using System.Threading.Tasks;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Extensions/AnimationExtension.cs]: usings outside namepace declaration

File: src/Animations/Animations/Helpers/EasingHelper.cs [hunk 23]
```diff
@@ -0,0 +1,38 @@
+﻿namespace FormsCommunityToolkit.Animations
+{
+    using Xamarin.Forms;
```
Comment (bartlannoeye) [inline src/Animations/Animations/Helpers/EasingHelper.cs]: usings outside namepace declaration

File: src/Animations/Animations/Animations/BounceAnimation.cs [hunk 24]
```diff
@@ -0,0 +1,82 @@
...
+            return Task.Run(() =>
+            {
+                Device.BeginInvokeOnMainThread(() =>
+                {
+                    Target.Animate("BounceIn", BounceIn(), 16, Convert.ToUInt32(Duration));
+                });
+            });
+        }
+
+        internal Animation BounceIn()
```
Comment (bartlannoeye) [inline src/Animations/Animations/Animations/BounceAnimation.cs]: Does this have to be internal or can it be private? (haven't opened the source to check references). Similar remark on the other animations.

File: src/Animations/Animations/Animations.csproj (hunk 1 above)
Comment (jsuarezruiz) [inline src/Animations/Animations/Animations.csproj]: Oops!.

File: src/Animations/Animations/Helpers/EasingHelper.cs (hunk 23 above)
Comment (jsuarezruiz) [inline src/Animations/Animations/Helpers/EasingHelper.cs]: Changed.

File: src/Animations/Animations/Behaviors/BeginAnimationBehavior.cs (hunk 21 above)
Comment (jsuarezruiz) [inline src/Animations/Animations/Behaviors/BeginAnimationBehavior.cs]: Good point.

File: src/Animations/Animations/Behaviors/BeginAnimationBehavior.cs [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+    public class BeginAnimationBehavior : Behavior<VisualElement>
+    {
+        private static VisualElement associatedObject;
+
+        protected override async void OnAttachedTo(VisualElement bindable)
+        {
+            base.OnAttachedTo(bindable);
+            associatedObject = bindable;
+
+            if (Animation != null)
```
Comment (Depechie) [inline src/Animations/Animations/Behaviors/BeginAnimationBehavior.cs]: Try to only resort to positive checks... in other words change this to: if(Animation == null) return;
Is one of the code guidelines ;)

Comment (jsuarezruiz) [inline src/Animations/Animations/Behaviors/BeginAnimationBehavior.cs]: Sounds good. Changed.

**Pull request merged**